lib/rspec/support/encoded_string.rb

```diff
@@ -112,11 +112,40 @@ def matching_encoding(string)
           string = remove_invalid_bytes(string)
           string.encode(@encoding)
         rescue Encoding::UndefinedConversionError, Encoding::InvalidByteSequenceError
-          string.encode(@encoding, ENCODE_UNCONVERTABLE_BYTES)
+          encode_unconvertable_bytes(string)
         rescue Encoding::ConverterNotFoundError
-          string.dup.force_encoding(@encoding).encode(ENCODE_NO_CONVERTER)
+          encode_no_converter(string.dup.force_encoding(@encoding))
         end
 
+        private
+
+        # On Ruby 2.7.0 keyword arguments mixed with conventional cause a warning to
+        # be issued requiring us to be explicit by using a ** to pass the hash as
+        # keyword arguments. Any keyword argument supporting Ruby supports this.
+        if RubyFeatures.kw_args_supported?
+          # Note on non keyword supporting Ruby ** causes a syntax error hence
+          # we must use eval. To be removed in RSpec 4.
+          binding.eval(<<-CODE, __FILE__, __LINE__)
```

Comment:
Thanks Jon for the patch.
That's sad we have to do that to deal with Ruby 2.7.

Reply:
Yeah I can't think of another away around it except having an extra file thats not required by the older build... I'm unsure if I like that less...

Reply:
Yes me too. The comments help for this part. 

Thanks

Reply:
While fixing 2.7 compatibility for aruba, I found you can still pass the keyword arguments as a 1.8-style hash literal. This would allow getting rid of the eval. You would also lose the constants though.

Reply:
@mvz I think that depends on the API in use, I tried the hash directly with `String#encode` and it still produces the warning.

Reply:
@JonRowe I'm not sure how that happened, but I just had to try as well and it seems to work here too: https://github.com/rspec/rspec-support/pull/394.

Reply:
🤦‍♂ 
```Ruby
2.7.0 > "abc".encode("UTF-8", {:invalid => :replace})
(irb):2: warning: Using the last argument as keyword parameters is deprecated
 => "abc"
2.7.0 > "abc".encode("UTF-8", :invalid => :replace)
 => "abc"
```

Reply:
🤦‍♂